### PR TITLE
Maps files having same paths to the same source label range

### PR DESCRIPTION
**Pull request by hbrodin**
This essentially makes polytracker view files having same paths being equal.

### CODE REVIEW

File: polytracker/src/taintdag/fdmapping.cpp [hunk 1]
```diff
@@ -1,5 +1,28 @@
 #include "taintdag/fdmapping.hpp"
 
+#include <iterator>
+
+namespace {
+
+// Helper structure that allows range-based for iteration
+// on the entries in FDMapping.
+template <typename It> struct RetVal {
```
Comment (surovic) [inline polytracker/src/taintdag/fdmapping.cpp:9]: We could maybe use [llvm::make_range](https://llvm.org/doxygen/iterator__range_8h.html) here?

**Review by surovic — APPROVED** [review]

File: polytracker/src/taintdag/fdmapping.cpp (hunk 1 above)
Comment (hbrodin) [reply polytracker/src/taintdag/fdmapping.cpp:9]: True. It is almost identical implementation, they handle more cases (move etc.). I don't think we have any dependency towards llvm in the runtime part of polytracker. But this is header only and is probably a good idea. Will have a look.

Comment (surovic) [reply polytracker/src/taintdag/fdmapping.cpp:9]: > I don't think we have any dependency towards llvm in the runtime part of polytracker.

Ah, okay, didn't realize this! I think we should keep it that way then.

Comment (hbrodin) [reply polytracker/src/taintdag/fdmapping.cpp:9]: We can't use it right away, would still need some code to map it into a reverse iterator. Keep as is?

Comment (surovic) [reply polytracker/src/taintdag/fdmapping.cpp:9]: Yes.

**Pull request merged**